### DISCUSSION TITLE
About version name convention

### DISCUSSION THREAD

**Discussion by JOTSR** [webui-dev/webui#137] [Q&A]
I would like to share my thoughts on the naming conventions of the languages port. As I mainly follow the Deno port, I have considered the readability of the version tag. As mentioned in [discussion](https://github.com/webui-dev/nim-webui/issues/13), I think that appending another number version directly to the main API version makes it hard to read and hard to distinguish the port version state from the API state. Moreover, it does not make it simple to inform users on port compatibility as there is no differentiation between major and minor version number updates (independently from the C API).

Therefore, I propose to use a [two-part semver inspired notation](https://www.drupal.org/docs/develop/git/git-for-drupal-project-maintainers/release-naming-conventions#release-tags) as follows:

```
{API version}-{port version}
```

- API version: major.minor.patch (full semver notation)
- port version: major.minor (reduced semver notation)

For example:
```
2.3.0-1.2 (webui@2.3.0, <port>-webui@1.2)
```

I think that this makes it more readable, easier to track and adds more flexibility for updating the portage.

Example with the Deno port:
```ts
// Current proposed version notation
import { WebUI } from 'https://deno.land/x/webui@2.3.0.4/mod.ts' // deno-webui v4
import { WebUI } from 'https://deno.land/x/webui@2.3.0.7/mod.ts' // deno-webui v7

// New version notation
import { WebUI } from 'https://deno.land/x/webui@2.3.0-1.2/mod.ts' // deno-webui v1.2
import { WebUI } from 'https://deno.land/x/webui@2.3.0-1.4/mod.ts' // deno-webui v1.4 (clearly not introduce breaking changes)
```

In the new version, the compatibility level is clear without having to check any docs compatibility table.

I would like to hear your opinion on this matter. Do you consider it a viable option to pursue? Please share your thoughts and feedback.

**Comment by AlbertShown** [comment]
Yes, Good suggestion 👍
This new version format keeps the condition with the [version discussion](https://github.com/webui-dev/nim-webui/issues/13), so I vote Yes!

**Comment by hassandraga** [comment]
LGTM
